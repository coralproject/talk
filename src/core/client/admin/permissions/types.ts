import { GQLUSER_ROLE_RL } from "coral-framework/schema";

export type PermissionMap<AbilityType extends string, ContextType> = {
  [abilityType in AbilityType]: {
    [role in GQLUSER_ROLE_RL]?: (context?: ContextType) => boolean;
  };
};
