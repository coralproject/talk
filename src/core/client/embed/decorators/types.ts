import pym from "pym.js";

export type CleanupCallback = () => void;
export type Decorator = (pym: pym.Parent) => CleanupCallback | void;
