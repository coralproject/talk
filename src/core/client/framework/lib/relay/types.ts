import { _RefType } from "react-relay";

export type FragmentKeys<T> = {
  [P in keyof T]: T[P] extends _RefType<any> | null ? P : never
}[keyof T];

export type FragmentKeysNoLocal<T> = Exclude<FragmentKeys<T>, "local">;
