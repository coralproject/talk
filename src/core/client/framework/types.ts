// TODO: (@cvle) Extract useful common types into its own package.
export { Diff, Omit, Overwrite, PropTypesOf } from "talk-ui/types";
