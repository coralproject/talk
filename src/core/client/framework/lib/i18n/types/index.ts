export { default as FluentShortNumber } from "./FluentShortNumber";
