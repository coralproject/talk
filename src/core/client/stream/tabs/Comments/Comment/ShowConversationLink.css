$commentsConversationLinkColor: var(--v2-colors-stream-blue-500);
$commentsConversationLinkColorHovered: var(--v2-colors-stream-blue-700);
$commentsConversationLinkColorActive: var(--v2-colors-stream-blue-800);

.conversationLink {
  &.colorRegular {
    color: $commentsConversationLinkColor;
  }

  &:not(.disabled) {
    &.colorRegular {
      &.mouseHover {
        color: $commentsConversationLinkColorHovered;
        & > span {
          text-decoration: underline;
        }
      }
      &.active,
      &:active {
        color: $commentsConversationLinkColorActive;
      }
    }
  }
}
