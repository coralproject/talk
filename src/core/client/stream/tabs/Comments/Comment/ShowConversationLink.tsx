import { Localized } from "@fluent/react/compat";
import cn from "classnames";
import React, {
  EventHandler,
  FunctionComponent,
  MouseEvent,
  useMemo
} from "react";

import { Button } from "coral-ui/components/v2";

import styles from "./ShowConversationLink.css";

export interface ShowConversationLinkProps {
  id?: string;
  href?: string;
  onClick?: EventHandler<MouseEvent>;
  className?: string;
}

const ShowConversationLink: FunctionComponent<
  ShowConversationLinkProps
> = props => {
  const classesOverride = useMemo(
    () => ({
      colorRegular: styles.colorRegular,
      active: styles.active,
      mouseHover: styles.mouseHover,
      disabled: styles.disabled
    }),
    [styles]
  );

  return (
    <Localized id="comments-showConversationLink-readMore">
      <Button
        id={props.id}
        className={cn(styles.conversationLink, props.className)}
        variant="textUnderlined"
        color="regular"
        classes={classesOverride}
        href={props.href}
        onClick={props.onClick}
        target="_parent"
        anchor
      >
        Read More of this Conversation >
      </Button>
    </Localized>
  );
};

export default ShowConversationLink;
