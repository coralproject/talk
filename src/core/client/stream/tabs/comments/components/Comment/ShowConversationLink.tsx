import React, { EventHandler, MouseEvent } from "react";
import { FunctionComponent } from "react";

import { Localized } from "fluent-react/compat";
import { Button } from "talk-ui/components";

export interface ShowConversationLinkProps {
  id?: string;
  href?: string;
  onClick?: EventHandler<MouseEvent>;
}

const ShowConversationLink: FunctionComponent<
  ShowConversationLinkProps
> = props => {
  return (
    <Localized id="comments-showConversationLink-readMore">
      <Button
        id={props.id}
        variant="underlined"
        color="primary"
        href={props.href}
        onClick={props.onClick}
        target="_parent"
        anchor
      >
        Read More of this Conversation >
      </Button>
    </Localized>
  );
};

export default ShowConversationLink;
