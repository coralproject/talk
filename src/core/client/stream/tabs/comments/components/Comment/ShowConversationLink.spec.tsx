import { noop } from "lodash";
import React from "react";
import { createRenderer } from "react-test-renderer/shallow";

import { PropTypesOf } from "talk-framework/types";

import ShowConversationLink from "./ShowConversationLink";

it("renders correctly", () => {
  const props: PropTypesOf<typeof ShowConversationLink> = {
    id: "id",
    onClick: noop,
    href: "http://localhost/comment",
  };
  const renderer = createRenderer();
  renderer.render(<ShowConversationLink {...props} />);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
