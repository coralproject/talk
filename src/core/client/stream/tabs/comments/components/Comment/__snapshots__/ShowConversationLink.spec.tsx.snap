// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Localized
  id="comments-showConversationLink-readMore"
>
  <ForwardRef(forwardRef)
    anchor={true}
    color="primary"
    href="http://localhost/comment"
    id="id"
    onClick={[Function]}
    target="_parent"
    variant="underlined"
  >
    Read More of this Conversation &gt;
  </ForwardRef(forwardRef)>
</Localized>
`;
